src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterHelper.cs

```diff
@@ -144,7 +144,13 @@ private static void SanityCheckBlob(object obj, string[] blobs)
                 string runtimeComparableBlob = CreateComparableBlobInfo(runtimeBlob);
 
                 Assert.True(storedComparableBlob == runtimeComparableBlob,
-                    $"The stored blob for type {obj.GetType().FullName} is outdated and needs to be updated.{Environment.NewLine}Stored blob: {blobs[frameworkBlobNumber]}{Environment.NewLine}Generated runtime blob: {runtimeBlob}");
+                    $"The stored blob for type {obj.GetType().FullName} is outdated and needs to be updated.{Environment.NewLine}{Environment.NewLine}" +
+                    $"-------------------- Stored blob ---------------------{Environment.NewLine}" +
+                    $"Encoded: {blobs[frameworkBlobNumber]}{Environment.NewLine}" +
+                    $"Decoded: {storedComparableBlob}{Environment.NewLine}{Environment.NewLine}" +
+                    $"--------------- Runtime generated blob ---------------{Environment.NewLine}" +
+                    $"Encoded: {runtimeBlob}{Environment.NewLine}" +
+                    $"Decoded: {runtimeComparableBlob}");
             }
```

Comment:
I thought about your suggestion and tbh I think to add that kind of logging in a precise manner (output all diff characters but not all if the whole string or a big part of it is different).

When these errors happen I always copy both blobs and compare them in my preferred diff checker. I don't think that adding more logging would eliminate that, therefore I will leave it as it is for now.